### DISCUSSION TITLE
Memory tweaks and proper Windows support

### DISCUSSION THREAD

**Discussion by kubilus1** [kubilus1/autoortho#3] [Announcements]
* Should work on Windows without WSL2 now.  Basically install WinFSP first.
* Tweaks to how we cache and handle tiles in memory
* Add caching of individual satellite images
* Fix some issues where multiple reads after header can occur but we aren't referencing mipmap0

For now, users should ignore the run.bat file and run this from the command prompt or powershell:
Install prereqs:
```
py -m pip install -r requirements.txt
py -m pip install memory-profiler
py .\autoortho\autoortho-winfsp.py <ORTHO PATH> <CUSTOM SCENERY MOUNT POINT>
```

Replacing `<ORTHO PATH>` with the ortho foler and `<CUSTOM SCENERY MOUNT POINT>` to the path in your XPlane custom scenery dir to mount this.

<hr /><em>This discussion was created from the release <a href='https://github.com/kubilus1/autoortho/releases/tag/0.0.5'>Memory tweaks and proper Windows support</a>.</em>

**Comment by kubilus1** [comment]
Currently this is in pre-release and needs some cleanup, but should work for most use cases.

**Comment by betampona** [comment]
On the last few posts you told us on one:
If on Windows run with
 py autoortho/autoortho-winfsp.py <orthopath> <custom scenery mount point>
and on another one :
py .\autoortho\autoortho-winfsp.py <ORTHO PATH> <CUSTOM SCENERY MOUNT POINT>
which one is the right one?

Currently I'm trying with both but I get the same error message [Errno 2] No such file or directory
Here is what I am typing:
py .\autoortho\autoortho-winfsp.py C:\users\dominique\ortho\z_ortho G:\X-Plane 11\Custom Scenery\z_autoortho
I'm on 2 different disks

Translated with www.DeepL.com/Translator (free version)

**Reply by ekc007** [reply]
I had the same problem. You have to try the second one and be at first in the right directory "cd C:\autoortho-0.0.5".
then you would use the second command but there is a typo..it should be "autoortho-**winfspy**.py". This way he finds the path but in my case the error pops up "module ... not found". So I tried to install (found this solution over the internet) winfsp module with typing "pip install winfspy". Then "py .\autoortho\autoortho-**winfspy**.py C:\users\dominique\ortho\z_ortho G:\X-Plane 11\Custom Scenery\z_autoortho"

Do this all before creating the directory "G:\X-Plane 11\Custom Scenery\z_autoortho". It will be created automatically and you can check it thereafter

**Reply by ekc007** [reply]
For me there were also problems with recognizing the paths if the is anywhere a space in between. in this case put the path in **"**path**"**

**Comment by betampona** [comment]
I went through the command prompt to get the right directory cd C:\autoortho-0.0.5
Then I installed the winfsp module by typing "pip install winfspy".
then I typed the paths py .\autoortho\autoortho- winfspy .py C:\users\dominique\ortho\z_ortho G:\X-Plane 11\Custom Scenery\z_autoortho paying attention to the possible spaces 
I did not create a z_autoortho in the custom scenery
see the picture I end up with the same error.
For path I do not know the manipulation 

Translated with www.DeepL.com/Translator (free version)
![winfspy](https://user-images.githubusercontent.com/97102929/195995761-46511bf4-d22a-4fbe-be93-0bae2d90dec3.JPG)

**Comment by kubilus1** [comment]
New pre-release version is available: https://github.com/kubilus1/autoortho/releases/tag/0.0.6  

This changes a few things around, but hopefully should be a bit easier to use:

* A configuration start screen is now present.
* The orthophotos directory config option should point to your extracted orthos directory
* XPlane scenery dir should point to the `Custom Scenery` dir in your Xplane install
* This will automatically attempt to setup dirs based on these locations and should warn you if something doesn't look right.
* An option to override the maptype is present.  So you could start with 'Arc' tiles and switch to 'BI' tiles.  
* Lots of tweaks and bug fixes.

Windows users should:
* Make sure they have WinFSP already installed
* Make sure they have an up to date Python installed (with pip)
* Extract the autorotho-0.0.6.zip and execute with 'run.bat'

Linux users should:
* Make sure fuse is installed and setup.
* Make sure an up to date Python and pip is installed
* Install pre-reqs: `python3 -m pip install -r requirements.txt`
* Run with `python3 autoortho`

So far my testing shows this works on Linux and Windows for both Xplane 11 and 12.  I'll do some more cleaning up before I make this a proper release.

**Comment by betampona** [comment]
Eureka it works, I did a test it looks really good on a BI 17 tile.
Can we have a button on the desktop to quickly launch the run.bat and fly when launching X-plane 11, this would avoid going to the disk.
I often start from the old arrival airport to a new destination, could we have an archive of the old location to make it quicker to launch x-plane?
You have done a great job that few people thought possible.

Translated with www.DeepL.com/Translator (free version)

**Reply by kubilus1** [reply]
Autoortho will cache the satellite imagery as it downloads it, which should make subsequent runs faster.  Currently I do not cache the XPLane formatted tiles, but that's something I can consider.  These take up a bit more space so I'll have to figure out how to make that efficient.

Glad it is working for you!

**Comment by ivotaHR** [comment]
1. PS C:\autoortho-0.0.6\autoortho> python3.exe .\autoortho.py
Windows detected
ChunkGetter got: 0
chunk_getter: <getortho.ChunkGetter object at 0x000001F32CCB4550>
Config file found C:\Users\Ivica\.autoortho reading...
Config file found C:\Users\Ivica\.autoortho reading...
Updating config.
{'orthos': 'C:/autoortho-0.0.6/autoortho/0/z_utah', 'Browse': '', 'scenery': 'C:/X-Plane 11/Custom Scenery', 'Browse0': '', 'showconfig': True, 'maptype': 'BI'}
{'general': {'# use gui config at startup': '', 'gui': 'True', '# show config setup at startup everytime': '', 'showconfig': 'True'}, 'paths': {'# ortho photos path': '', 'orthos_path': 'C:/autoortho-0.0.6/autoortho/0/z_utah', '# x-plane custom scenery path': '', 'scenery_path': 'C:/X-Plane 11/Custom Scenery'}, 'autoortho': {'# override map type with a different source': '', 'maptype_override': 'BI'}}
Wrote config file: C:\Users\Ivica\.autoortho
Wrote config file: C:\Users\Ivica\.autoortho
Config file found C:\Users\Ivica\.autoortho reading...
Config file found C:\Users\Ivica\.autoortho reading...
Preparing directory structures....
Diretories are ready!
root: C:/autoortho-0.0.6/autoortho/0/z_utah\textures
mountpoint: C:/X-Plane 11/Custom Scenery\z_autoortho\textures
Running in Windows WinFSP mode.
Started tile clean thread.  Mem limit 2147483648
Started tile clean thread.  Mem limit 2147483648
NUM TILES CACHED: 0.  TOTAL MEM: 52 MB
NUM TILES CACHED: 0.  TOTAL MEM: 52 MB
TILE CACHE:  MISS: 0  HIT: 0
TILE CACHE:  MISS: 0  HIT: 0
Starting FS
Stopping FS
FS stopped
Traceback (most recent call last):
  File "C:\autoortho-0.0.6\autoortho\autoortho.py", line 569, in <module>
    main()
  File "C:\autoortho-0.0.6\autoortho\autoortho.py", line 544, in main
    autoortho_winfsp.main(root, mountpoint,
  File "C:\autoortho-0.0.6\autoortho\autoortho_winfsp.py", line 867, in main
    fs.start()
  File "C:\Users\Ivica\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\winfspy\file_system.py", line 172, in start
    raise WinFSPyError(f"Cannot mount file system: {cook_ntstatus(result).name}")
winfspy.plumbing.exceptions.WinFSPyError: Cannot mount file system: STATUS_INVALID_PARAMETER
PS C:\autoortho-0.0.6\autoortho>

**Reply by ErikSchlijper** [reply]
How did you fix this? I'm having the exact same issue!

**Comment by ivotaHR** [comment]
It is working!!!
Thank you  We need Global scenery pack.

**Reply by kubilus1** [reply]
That's exactly what I've been working on ;)

**Comment by betampona** [comment]
Hello, I have a test in IFR with the zibo between Nice and Paris in BI zoom 17, I had misses of tiles due to the speed of the plane compared to the reloading of the tiles although I have a flow of 60 mb/s.I am doing a test on France in BI 15 without the zoom 18 near the airports, for the moment I am downloading on Ortho4xp, then I will make you a return.For VFR I think it's okBut for IFR you need a zoom level lower than 17 unless you find a way to make the zoom fluctuate according to altitude and speed.Greetings.Translated with www.DeepL.com/Translator (free version)envoyé : 16 octobre 2022 à 15:26de : ivotaHR ***@***.***>à : kubilus1/autoortho ***@***.***>cc : betampona ***@***.***>, Comment ***@***.***>objet : Re: [kubilus1/autoortho] Memory tweaks and proper Windows support (Discussion #3) It is working!!! Thank you We need Global scenery pack.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

**Reply by kubilus1** [reply]
I found that zoom level 16 with a higher airport zoom level is a good compromise between speed and quality.  Particularly at 1080p resolutions.

This tool already will get the minimum possible amount of data dynamically by effectively retrieving only the resolution of data that XPLane requests at any given time.  If you are flying higher, or something is far, you don't get the full resolution of a tile.

At higher zoom levels, you effectively have many more tiles per area, which means more time and downloading will occur.  This very well may be waste in a lot of cases.  XPlane will only actually read in a limited amount of data per tile at any given time.  You can see this in the XPlane developer tools tile explorer.  It will indicate the resolution it actually has read in.  4096x4096 is full resolution, but often you see much lower.  If your graphics settings are not maxed out for textures you may *never* actually read in full texture resolution of a tile.

60mb/s is on the slow side for internet, that combined with high ZL (ZL17) and perhaps you are far from the tile servers.  Your internet may also have latency that isn't measured in the download speed.  Autoortho will cache your downloaded tiles though, so subsequent runs should be a bit faster.

Another factor to consider is your CPU.  This tool will generate the format XPlane needs on the fly.  A slower CPU will mean slower processing of new tiles and possible delay.

**Comment by ivotaHR** [comment]
When we fly  ,does the cache (2GB  usage-limit) automated deleted, updated??

**Reply by kubilus1** [reply]
The 2GB memory limit is a target for retention of the on-the-fly processed tiles.  It will automatically attempt to remove tiles when it can.  The on-disk satellite imagery cache though does not automatically get cleaned up.

**Comment by ivotaHR** [comment]
For me ZL16 is top of my choice.  No need better. I am on 1080p. I can use due data traffic, cpu usage ZL15 or less.

**Comment by betampona** [comment]
I just did a test in BI15 with the airports in ZL18 (I thought I had removed them)
Overall the flight went well LFMP LFPG 500 nautical miles flight time 1h20, I was with the zibo at FL 360 ground speed 480kts.
I had a few pale blue tiles on the right side of the screen at half height, limit between the near and far scenery.
By continuing the flight this blue tile disappeared and the scenery came back.
On the right side no problem.
Do you have any idea about this problem?
The quality of the BI15 is suitable for high altitude flights. 

My fibre at the box is 90 Mbit/s or 11.25 Mb/s but my computers are in the attic and I have 70 Mbit/s or 8.75 Mb/s through PLCs
In France there are 2 fibre contracts, the first one is 400 Mbit/s but in reality we have 100 Mbit/s, the second one is 1Gbit/s but it must be much more expensive and I have strong doubts about the speed.

Translated with www.DeepL.com/Translator (free version)

**Reply by kubilus1** [reply]
The blue areas are basically what XPlane does when it is waiting to read a DSF file.  You will get the same result if you try to load orthos from a slow hard drive, for instances.

I'm not sure a fully understand how you are describing the computer in your attic, but if this is over a slow local network, that could cause extra delays, and perhaps the blue tiles you saw.

**Comment by betampona** [comment]
Hi kubilus1, you specify us option in run.bat:
An option to override the maptype is present. So you could start with 'Arc' tiles and switch to 'BI' tiles
what is the advantage of this option?

**Reply by kubilus1** [reply]
This allows you to try out different tile set types.  BI is from Bing maps.  Arc is from ArcGIS, etc.

**Comment by betampona** [comment]
You managed to make autoortho work under windows, what made you progress in the realization of the project compared to this summer when everything was blocked, you always used winfsp and python ?

**Reply by kubilus1** [reply]
So there are a lot of little reasons.  To start, understand that Linux and similar operating systems have the concept of userspace filesystems through a library called FUSE.  This makes it very simple to interact with a filesystem and create filesystem for various purposes.  This is the technique that this project is based off of.

WinFSP provides a somewhat analogous tool for Windows, but works differently.  It is advertised though to have a FUSE compatibility layer.  This is initially what I tried out for Windows compat.  It works in some modes, such as with file explorer and editing tools but for a variety of little reasons XPlane would refuse to recognize WinFSP-FUSE file system mounts.

So it is possible to write against WinFSP at a lower level, but that is a lot more work, and typically done in C or C++.  Which I can program, but that's a lot lot more effort.  Plus I don't use Windows, so it's not a huge amount of value to me.  I also did not have much of a system to test out Windows changes.

I discovered a project though that wraps WinFSP for python and determined it may be feasible to create a new layer using this just for Windows users.  This is the current implementation.  Plus I found a little bit of time and energy to work on it.

**Comment by betampona** [comment]
Thanks for the reply, I have just switched to 1Gb/s fibre instead of 100Mb/s, I have gained in launch time from 4mn to 2mn40, I am in ZL15 with ZL18 on the airports. I'm going to test the scenery, see if I don't have any more blue tiles synonymous with delay on the scenery. As I'm flying at 480 kts or 900 km/h at FL from 300 to 380 the tiles renewal are very close in time. Last time with the old fibre the blue tiles were at minimum.
Greetings.

Translated with www.DeepL.com/Translator (free version)

**Comment by betampona** [comment]
For the flight no problem of downloading tiles in BI 15 for IFR that seems to me good, to test BI16.
I have attached a picture of the download speed and periodicity, would you be interested?
![temps de chargement autoortho](https://user-images.githubusercontent.com/97102929/196969653-8f39a1c1-8ffc-4ab8-8073-f5bee43bec8d.jpg)

**Comment by kubilus1** [comment]
If anyone is willing to try, I branched off ortho creation into a separate repo.  Here's a release that is equivalent to the Pacific/Australia global scenery DLC pack: https://github.com/kubilus1/autoortho-scenery/releases/tag/0.0.10

This is in several zip file chunks, so some assembly is required!

**Comment by betampona** [comment]
Hello, I downloaded z_aus_pac_00 and tried to view it on ortho4xp but it is not viewable on ortho4xp on the world map, the 7 folders represent the whole of Australia?
To compile them I use the file manager method and in word search I type: terrain, texture, Earth nav data and it compiles everything into each folder and I end up with 3 final folders.
Is there another method?

**Reply by kubilus1** [reply]
Yes, the files are in the layout needed for XPlane, not for Ortho4XP so they won't show up in the Ortho4XP world map.  Yes, combining the folders for each z_aus_pac_##.zip sounds correct.

This covers the entire Pacific, Australia, Oceania region so that would include Australia, NewZealand, Indonesia, and many other places.

**Comment by betampona** [comment]
I did the compilation of z_ it was faster than I thought with only 7 folders it makes the work easier, in general with the field folder, small in size but big in information we spend time.
After compilation I have 6,65Go.
At the first launch it took me 6 mn, by looking at the log.txt of x-plane it loaded 12 tiles at the beginning, I made an attempt of second launch and there it took 1mn.
I did a flight from Canberra YSCB to Brisbane YBBN about 550 nautical miles at 430kts at FL380, visually it was very nice especially at the arrival.
I still have your blue tile start on the sides but it is fading, this is probably due to a delay in the download.
I should do a test with Earth nav data and terrain installed in z_ortho, see if it is possible.
I would be interested in Europe up to the Urals with the Mediterranean and the Canary Islands, Madeira, Azores, Iceland and the Faroes.
Best regards.

Translated with www.DeepL.com/Translator (free version)

**Reply by kubilus1** [reply]
So one thing I have noticed.  XPlane 12 seems to be more efficient with how it loads DSF files.  I notice much less often seeing 'blue regions' like you describe on XPlane 11.

**Comment by betampona** [comment]
Hello, is there a program change between autoortho-0.0.6 the one installed on my computer and this latest version autoortho-0.1.0 to download?

**Reply by kubilus1** [reply]
Not really, they are basically the same with some very minor code cleanup that should not affect functionality.

**Comment by betampona** [comment]
Hello, I am still in version 0.0.6, my last tests are done now with Earth nav data and hard terrain in z_ortho in custom scenery, my last flight of 800 nautical miles from YPDN to YBAS speed 435kts FL370 I have an appearance from time to time of the blue tile that fades quickly, I even had a disappearance of the background scenery that returned after a while, the scenery of the front was always present.
I would have liked an explanation on the cache, it keeps in memory the downloaded tiles until what height of memory?
 Where is this cache located?
A tile is 10.6 MB I think.
It would be possible for you to create an Orthophoto zone to see the photo, that's where I'm mainly flying.
Greetings

Translated with www.DeepL.com/Translator (free version)
![Carte EUR NA](https://user-images.githubusercontent.com/97102929/198869369-1c2eead0-6c56-4833-b588-58a3402f0c61.JPG)

**Reply by kubilus1** [reply]
Effectively, there are multiple caches.  It's worth considering a few distinct components of how XPlane access terrain and textures.
* A scenery file with extension .DSF file contains an area of information about terrain and textures.  This may references to many texture files.  The size of these can vary quite a bit.
* The texture files, also referred to as 'tiles', typically have an extension of .DDS and represent 4096x4096 resolution data with multiple mipmaps.  The full-size of a DDS file is just over 22MB.
* Satellite image providers use a 'slippymap' format that returns 256x256 either PNG or JPEG files.  So it takes 256 of these photos to make up a full-size DDS file.

In memory I retain a certain amount of these processed DDS Tiles.  On the filesystem I cache all downloaded satellite photos in the autoortho directory (currently though this is subject to change).

And yes, I plan to create scenery for Europe region eventually.

**Comment by betampona** [comment]
Hi, I wanted to upgrade to autoortho0.1.0, I got an error message, see picture.
![error run bat](https://user-images.githubusercontent.com/97102929/199273521-86dda699-8c4a-4c31-8b97-c2e5ff412de8.JPG)

**Comment by kubilus1** [comment]
The message definitely works.   The scenery path should not contain
'z-autoortho'

**Comment by ivotaHR** [comment]
Upgrade to autoortho0.1.0 works well for me. 
Question: is possible to make 'rolling cache'  like msfs2020 features?
Thank you.

**Reply by kubilus1** [reply]
It's definitely possible to have a rolling cache.  I'll probably do this at some point.  Glad it's working for you.

**Comment by betampona** [comment]
I've already changed the name, I put any name in custom scenery and it gives me the same error message.
I'm going to do a clean install of python3 again unless you have another method.

**Reply by kubilus1** [reply]
The custom scenery folder you specify should not have any extra names in it.  It should end in 'Custom Scenery'.

**Comment by betampona** [comment]
Hi, I understand about the folder in the custom scenery.
My problem now is the launch of Run Bat.
See the picture before shutdown.
the history:
1) I deleted python3
2) delete version autoortho0.06
3) I downloaded the python3.11 version
4) downloaded autoortho0.1.0
5) tried to run pip
6) restarted winfsp
7) downloaded visual studio
Run.bat doesn't want to give the programming image of the files.
![run bat 1](https://user-images.githubusercontent.com/97102929/199693104-4815f352-4748-4bb3-ab78-59a647831a36.JPG)

**Reply by kubilus1** [reply]
Looks like you need to install some Microsoft thing.  I'd try the link they suggest.

**Reply by kubilus1** [reply]
It's probably requiring the build since you are using a cutting edge new release of Python.  I'll bet this was prebuilt previously.  You could try going back to Python 3.10, but I hear 3.11 has performance improvements.

**Comment by betampona** [comment]
I went back to the version of python3.10.8
Here is the picture of the run.bat before shutdown.
![run bat2](https://user-images.githubusercontent.com/97102929/199771088-2dc5903c-9320-4f54-a71a-3898362fb7dc.JPG)

**Comment by kubilus1** [comment]
See if there is a log file in the same directory you ran that from.

**Comment by betampona** [comment]
When I run run.bat I get an autoortho.log file that appears in autoortho-0.1.0
My problem with run.bat started when I removed python 3.10.8 and reinstalled it.

**Reply by kubilus1** [reply]
Did you see anything in the log file?  I would make sure you still have WinFSP setup.  The run.bat is pretty basic, you could try manually executing those commands and see what happens.

**Comment by betampona** [comment]
After cleaning, I re-downloaded and installed everything, see photo.
I looked under C: with powerShell the version of python 3.10.8 and pip 22.3
I'm not able to access the implementation interface with run.bat.

![run bat 3](https://user-images.githubusercontent.com/97102929/200158838-a8e4d5cf-4de9-4690-ba70-4f4286abf1cb.JPG)

**Comment by betampona** [comment]
If you are talking about autoortho.log it is empty, as for the run.bat the python screen does not stay on the screen.When I go to the autoortho folder of autoortho-0.1.0 and click on autoortho.py I launch the autoortho.setup image, I enter the direction of the ortho folder and the direction of the custom scenery but when I click on Fly the python image disappears, on the other hand I have the creation of an .autoortho or in the paths part I have the directions saved, but as the python screen does not stay in run.bat when I launch x-plane 11 there is no download.envoyé : 6 novembre 2022 à 16:38de : kubilus1 ***@***.***>à : kubilus1/autoortho ***@***.***>cc : betampona ***@***.***>, Comment ***@***.***>objet : Re: [kubilus1/autoortho] Memory tweaks and proper Windows support (Discussion #3) Did you see anything in the log file? I would make sure you still have WinFSP setup. The run.bat is pretty basic, you could try manually executing those commands and see what happens.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

**Reply by kubilus1** [reply]
Instead, try to open 'powershell' go to the directory you have extracted autoortho and type the commands that are in the run.bat file.

First:
```py -m pip install -r requirements-win.txt```

Followed by:
```py autoortho```

**Comment by betampona** [comment]
I typed the 2 command lines in autoortho-0.1.0 see the picture.
run.bat is still not active

![run bat 4](https://user-images.githubusercontent.com/97102929/200237406-df23eb44-11f1-4be5-8382-3a972a216cc9.JPG)

**Reply by kubilus1** [reply]
Okay, this is helpful, I think I see the issue.  The quick fix is for you to just do a `py -m pip install fusepy`.

**Comment by betampona** [comment]
On the attached picture I typed in powershell the direction to run.bat.
it appears below C:\>py -m pip install -r requirements-win.txt an error line in red, ERROR: Cloud not open requirements file: Errno 2 No such file or directory: "requirements-win.txt" 
is this normal?

![run bat 5](https://user-images.githubusercontent.com/97102929/200250355-46a96a6b-1e69-4fe0-9585-d40c0b6898a4.JPG)

**Comment by betampona** [comment]
py -m pip install fusepy solved the problem, I did a test everything is ok.

**Comment by betampona** [comment]
Hello, I have just looked at the download speed of the scenery during a first flight using the task manager. I never exceed 25MBits/s and it takes about 6 minutes to launch the scenery in x-plane 11, yet in fibre I have 300Mbits/s. Where can this limitation come from?
If you do the same test with the task manager what value do you have?

**Reply by kubilus1** [reply]
I suspect the bottleneck is the satellite image service itself (bing maps, ArcGIS, or whatever else).  It's possible CPU is your limit.  A test you can do is to re-load the same flight.  For me this will take about 5-6 min the first time, but then about 1-2 min following times.  If you see it taking around 6 min every time and your CPU is maxed, then you are CPU bound.

**Comment by ivotaHR** [comment]
https://imgur.com/a/8CXukOV

TIP:
Is possible to make tool like this for make ortho dsf files (to collect files from route)? Or we should wait full world ortho dsf scenery package.

**Reply by kubilus1** [reply]
So Ortho4XP can make these of course but you have to ahead of time.  It's also a pretty complex and difficult to use tool for beginners.

**Comment by betampona** [comment]
I am in the same case as you, I put 5 to 6 mn the first time then 1 mn if the cache is loaded. I am away from home for the moment but there is a command in powershell to optimize the download (internet search)Envoyé depuis mon appareil Galaxy
-------- Message d'origine --------De : kubilus1 ***@***.***> Date : 12/11/2022  02:33  (GMT+01:00) À : kubilus1/autoortho ***@***.***> Cc : betampona ***@***.***>, Comment ***@***.***> Objet : Re: [kubilus1/autoortho] Memory tweaks and proper Windows support (Discussion #3) 
I suspect the bottleneck is the service itself.  It's possible CPU is your limit.  A test you can do is to re-load the same flight.  For me this will take about 5-6 min the first time, but then about 1-2 min following times.  If you see it taking around 6 min every time and your CPU is maxed, then you are CPU bound.

**Reply by kubilus1** [reply]
Yes, this is not surprising then.  I am working on a change that should reduce the amount of downloaded data a bit more as well, so perhaps could speed this up somewhat, but don't expect anything too dramatic.

**Reply by kubilus1** [reply]
Here's a pre-prelease version that should somewhat reduce the downloaded data: https://github.com/kubilus1/autoortho/releases/tag/0.1.2

**Reply by ivotaHR** [reply]
> Here's a pre-prelease version that should somewhat reduce the downloaded data: https://github.com/kubilus1/autoortho/releases/tag/0.1.2

I tested it seems all ok.

**Comment by betampona** [comment]
I compared the downloads of version 01.0 and 0.1.2
the times are taken when run.bat starts to load.
On Alice Springs YBAS version 0.1.0, 1st download 5'33" to get the scenery when the cache is created 53" to get the scenery.
for version 0.1.2, 1st download 5'17" for the scenery and 46" when the cache is created.
I noticed that the number of tiles in the cache has decreased.
The best would be to have the possibility to download only the tiles in jpeg format of 256 by 256
A remark on ortho4xp the jpeg tiles are 4096x4096 and on autoortho they are 16 times smaller.

Translated with www.DeepL.com/Translator (free version)

**Reply by kubilus1** [reply]
Thank you for testing that.  So basically a 5% speed improvement when downloading images from scratch and a 13% improvement when already cached.  That's about what I expected.  The only really practical way to decrease downloads more would be to reduce the ZL that is used.

Your airport ZL in particular, since you start on the ground, you typically are downloading the maximum size.

I think it may be technically possible to speed up the tile processing as well, but probably won't make much difference.  Some time is taken by XPlane even if you don't use any of this stuff, afterall.

**Comment by betampona** [comment]
I did a test with ZL13 on Alice Springs it takes 22 seconds to get the set and 17" if the cache is loaded.
I made an attempt by mixing ZL16+18 and ZL13 in the same folder, I got the scenery in 22" autoortho stayed there, it did not load ZL16+18 because it started with the 12 tiles in ZL13 and as they were present in x-plane11 it blocked the other part of the download.
Yet it would be best to start in a degraded ZL13 form and then finish the download in ZL16+18

Translated with www.DeepL.com/Translator (free version)

**Comment by betampona** [comment]
Flight between YBAS and YPED FL370 speed 450 kts
as soon as I start at low altitude I have the appearance of a multitude of checkerboards on the ground very close to the plane, I no longer have the scenery in ZL16 with a certain depth, the scenery in ZL16 is just in front of me afterwards it's blurrier and in the distance it's blurred but it's normal on x-plane.
At FL370 level see the pictures, in front of me ZL16 in sawtooth on very little ground surface, the ZL16 should go to the normal limit allowed by x-plane.

Translated with www.DeepL.com/Translator (free version)
![b738 - 2022-11-18 06 50 16](https://user-images.githubusercontent.com/97102929/202635692-95c1db36-07df-405a-aefe-d429c961d84b.png)
![b738 - 2022-11-18 06 49 06](https://user-images.githubusercontent.com/97102929/202635720-7ac7c12b-99ff-493a-a60a-0798b465c922.png)

**Reply by kubilus1** [reply]
What versions are you using?  Probably from the source satellite photos.  Take a look at your map source and zoom in and out and you'll probably see a similar effect.  I could possibly make the minimum fetched zoom level tunable.

**Comment by betampona** [comment]
For orthophotos I use your AUS BI16 pack
Here is a picture at FL370 with the autoortho-0.1.2 version
![b738 - 2022-11-20 04 55 58](https://user-images.githubusercontent.com/97102929/202886023-add7b71a-1718-449f-87c3-d51eea255dde.png)

The following photo is with autoortho-0.1.0 at the same height and about the same place, there is no checkerboard and the ZL16 fills well its zone which is dedicated to him, it is much more pleasant visually even during the rise, it is of the traditional orthophoto that one finds on x-plane.

![b738 - 2022-11-20 05 25 33](https://user-images.githubusercontent.com/97102929/202886205-35c14dbf-c095-4cd1-a230-fe59b5910c21.png)

**Reply by kubilus1** [reply]
Thanks for that comparison.  Easy fix, effectively ver 0.1.2 allowed a min fetched ZL of 11 where 0.1.0 allowed a min ZL of 12.  I think we'll want 12 to be the default.

**Reply by betampona** [reply]
Have you released a new version?

**Comment by betampona** [comment]
I forgot to tell you that on the 1st photo autoortho-0.1.2 the undercoat is sweating on the ZL16 layer.

**Comment by betampona** [comment]
Hello, I see you are working on the code.
What are these changes?
Can we expect a new version soon?

**Reply by ivotaHR** [reply]
I am looking for new version , too.

**Reply by kubilus1** [reply]
Good stuff is on the way.  Just doing a little final testing and cleanup.  This will be a fairly significant update:
* GUI based ortho package downloader and installer
* Replacing the DDS file creation tooling (2X speed up when processing files!)
* Re-organizing code, etc
* New satellite source choices

**Comment by kubilus1** [comment]
FYI new release has been cut!

See: https://github.com/kubilus1/autoortho/releases/tag/0.2.0
Discussion thread: https://github.com/kubilus1/autoortho/discussions/20
